dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResult.java

```diff
@@ -0,0 +1,374 @@
+
+    public String getFolderId() {
+        return folderId;
+    }
+
+    public String getParentStr() {
+        return this.parent;
+    }
+
+    public NavResult(String parent, String hostId, String folderId, Long languageId) {
+        this.hostId = hostId;
+        this.folderId = folderId;
+        this.parent = parent;
+        this.languageId = languageId;
+
+
+        title = href = "";
+        order = 0;
+
+        try {
+            sysuser = APILocator.getUserAPI()
+                .getSystemUser();
+        } catch (DotDataException e) {
+            Logger.warn(this, e.getMessage(), e);
+        }
+
+    }
+
+
+
+    public NavResult(String parent, String host, Long languageId) {
+        this(parent, host, null, languageId);
+    }
+
+    public String getTitle() throws Exception {
+        if (title.startsWith("$text.")) {
+            ViewRenderTool render = new ViewRenderTool();
+            render.setVelocityEngine(VelocityUtil.getEngine());
+            return render.eval(title);
+        } else {
+            return title;
+        }
+    }
+
+    public String getHostId() {
+        return hostId;
+    }
+
+    public void setHostId(String hostId) {
+        this.hostId = hostId;
+    }
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+
+    public String getHref() {
+        return href;
+    }
+
+    public boolean isShowOnMenu() {
+        return showOnMenu;
+    }
+
+    public void setShowOnMenu(boolean showOnMenu) {
+        this.showOnMenu = showOnMenu;
+    }
+
+    public void setHref(String href) {
+        this.href = href;
+
+    }
+
+    public boolean isCodeLink() {
+        return this.codeLink != null;
+    }
+
+
+    public String getCodeLink() {
+        return codeLink;
+    }
+
+    public void setCodeLink(String codeLink) {
+        this.codeLink = codeLink;
+    }
+
+
+
+    public int getOrder() {
+        return order;
+    }
+
+    public void setOrder(int order) {
+        this.order = order;
+    }
+
+
+
+
+    public void setChildren(List<NavResult> children) {
+        this.children = children;
+    }
+
+    public boolean isFolder() {
+        return folderId != null;
+    }
+
+    public List<? extends NavResult> getChildren() throws Exception {
+        if (children == null && hostId != null && folderId != null) {
+            // lazy loadinge children
+            Host host = APILocator.getHostAPI()
+                .find(hostId, sysuser, true);
+            Folder folder = APILocator.getFolderAPI()
+                .find(folderId, sysuser, true);
+            Identifier ident = APILocator.getIdentifierAPI()
+                .find(folder);
+            NavResult lazyMe = new NavTool().getNav(host, ident.getPath(), languageId, sysuser);
+
+            children = lazyMe.getChildren();
+            childrenFolderIds = lazyMe.getChildrenFolderIds();
+        }
+        if (children != null) {
+            ArrayList<NavResult> list = new ArrayList<NavResult>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)